core/src/main/kotlin/com/strumenta/kolasu/transformation/Transformation.kt

```diff
@@ -17,15 +20,26 @@ annotation class Mapped(val path: String = "")
 /**
  * Factory that, given a tree node, will instantiate the corresponding transformed node.
  */
-class NodeFactory<Source>(
-    val constructor: (Source, ASTTransformer, NodeFactory<Source>) -> Node,
+class NodeFactory<Source, Output : Node>(
+    val constructor: (Source, ASTTransformer, NodeFactory<Source, Output>) -> Output,
     val children: MutableMap<String, ChildNodeFactory<Source, *, *>?> = mutableMapOf()
 ) {
+
+    private var finalizer: (Output) -> Unit = {}
+
+    fun finally(node: Output) {
```

Comment:
If I understand correctly this invoke the finalizer, while the following method with the same name set it. That could be maybe confusing

Reply:
See discussion below.

Reply:
We discussed this in a meeting and Lorenzo suggested renaming this one `finalize`